packages/fxa-shared/subscriptions/configuration/helpers.ts

```diff
@@ -0,0 +1,164 @@
+    support: ProductConfig['support'];
+  };
+
+  const planConfiguration = getPlanProductConfig(plan);
+
+  const availableLocales =
+    (planConfiguration.locales && Object.keys(planConfiguration.locales)) || [];
+  availableLocales.push(DEFAULT_LOCALE);
+  const selectedLocale = parser.pick(availableLocales, userLocales.join(','));
+
+  output = {
+    uiContent: planConfiguration.uiContent,
+    urls: planConfiguration.urls,
+    support: planConfiguration.support,
+  };
+
+  if (
+    planConfiguration.locales &&
+    selectedLocale &&
+    planConfiguration.locales[selectedLocale]
+  ) {
+    const localesConfig = planConfiguration.locales[selectedLocale];
+    output = {
+      uiContent: {
+        ...output.uiContent,
+        ...localesConfig.uiContent,
+      },
+      urls: {
+        ...output.urls,
+        ...localesConfig.urls,
+      },
+      support: {
+        ...output.support,
+        ...localesConfig.support,
+      },
+    };
+  }
+
+  return output;
+};
+
+export const urlsFromProductConfig = (
+  plan: Plan,
+  userLocales: readonly string[] = [DEFAULT_LOCALE],
+  useFirestoreProductConfigs: boolean
+) => {
+  if (useFirestoreProductConfigs) {
+    return flattenProductConfig(plan, userLocales).urls;
+  } else {
+    const planMetadataConfig = productDetailsFromPlan(plan, userLocales);
+
+    return {
+      termsOfService:
+        planMetadataConfig.termsOfServiceURL ||
+        DEFAULT_PRODUCT_DETAILS.termsOfServiceURL!,
+      termsOfServiceDownload:
+        planMetadataConfig.termsOfServiceDownloadURL ||
+        DEFAULT_PRODUCT_DETAILS.termsOfServiceDownloadURL!,
+      privacyNotice:
+        planMetadataConfig.privacyNoticeURL ||
+        DEFAULT_PRODUCT_DETAILS.privacyNoticeURL!,
+      privacyNoticeDownload:
+        planMetadataConfig.privacyNoticeDownloadURL ||
+        DEFAULT_PRODUCT_DETAILS.privacyNoticeDownloadURL!,
+      cancellationSurvey: planMetadataConfig.cancellationSurveyURL,
+      download:
+        plan.product_metadata?.downloadURL || plan.plan_metadata?.downloadURL,
+    };
+  }
+};
+
+export const webIconConfigFromProductConfig = (
+  plan: Plan,
+  userLocales: readonly string[] = [DEFAULT_LOCALE],
+  useFirestoreProductConfigs: boolean
+) => {
+  if (useFirestoreProductConfigs) {
+    const planConfiguration = getPlanProductConfig(plan);
+    const urls = flattenProductConfig(plan, userLocales).urls;
+
+    return {
+      webIcon: urls.webIcon,
+      webIconBackground: planConfiguration.styles?.webIconBackground,
+    };
+  } else {
+    const combinedMetadata = metadataFromPlan(plan);
+
+    return {
+      webIcon: combinedMetadata.webIconURL,
+      webIconBackground: combinedMetadata.webIconBackground,
+    };
+  }
+};
+
+export const uiContentFromProductConfig = (
+  plan: Plan,
+  userLocales: readonly string[] = [DEFAULT_LOCALE],
+  useFirestoreProductConfigs: boolean
+) => {
+  if (useFirestoreProductConfigs) {
+    return flattenProductConfig(plan, userLocales).uiContent!;
```

Comment:
Feels a little strange that you'd need to use `!` to assert something on a return type of a function that you created in this patch.  Why not update the return type itself?  I believe the TS inferred type should just work.

Reply:
Whoops, I thought I'd removed all of those.